File: UVa/Contents/d255/d255.cpp
#include<iostream>

# define MAX 501

using namespace std;

void init(void);

int phi[MAX] = {0};
int G[MAX] = {0};

int main(void)
{
	int N;

	init();
	while(1)
	{
		cin >> N;
		
		if(!N)
			break;
		
		cout << G[N] << endl;
	}
		
	
	return 0;
}

void init()
{	
	for(int i = 2; i < MAX; i++)
		phi[i] = i;
	
	for(int i = 2; i < MAX; i++)
	{
		if(phi[i] == i)
			for(int j = i; j < MAX; j += i)
				phi[j] = phi[j] / i * (i - 1);
		
		for(int j = 1; j * i < MAX; j++)
			G[j * i] += j * phi[i];
	}
	
	for(int i = 3; i < MAX; i++)
		G[i] += G[i - 1];
}

File: UVa/Contents/d255/d255.py
import sys

MAX = 501
phi = [0] * MAX
G = [0] * MAX
def init():
    for i in range(2, MAX):
        phi[i] = i
    
    for i in range(2, MAX):
        if phi[i] == i:
            j = i
            while j < MAX:
                phi[j] = phi[j] // i * (i - 1)
                j += i
        
        j = 1
        while j * i < MAX:
            G[j * i] += j * phi[i]
            j += 1

    for i in range(3, MAX):
        G[i] += G[i - 1] 


init()
for n in sys.stdin:
    n = int(n)

    if n == 0:
        break

    print(G[n])
